src/expr/node.h

```diff
@@ -1438,6 +1444,13 @@ NodeTemplate<ref_count>::substitute(Iterator substitutionsBegin,
   return substitute(substitutionsBegin, substitutionsEnd, cache);
 }
 
+template <bool ref_count>
+inline Node NodeTemplate<ref_count>::substitute(
+    std::unordered_map<TNode, TNode, TNodeHashFunction>& subs) const
+{
+  return substitute(subs.cend(), subs.cend(), subs);
```

Comment:
I don't follow how `subs.cend()` is used here, can you explain?

Reply:
Since all substitutions are already cached for us, we don't need to iterate over them. I picked `subs.cend()` arbitrarily. I could have picked `subs.cbegin()`. Added a comment to explain this.

Reply:
Ok sounds good, indeed it is somewhat arbitrary (or you could even make a dummy map).  But this is fine as long as its documented.

Reply:
Done.